de/nmichael/efa/core/config/EfaCloudUserRecord.java

```diff
@@ -0,0 +1,135 @@
+package de.nmichael.efa.core.config;
+
+import de.nmichael.efa.core.items.IItemListener;
+import de.nmichael.efa.core.items.IItemType;
+import de.nmichael.efa.data.storage.DataKey;
+import de.nmichael.efa.data.storage.DataRecord;
+import de.nmichael.efa.data.storage.IDataAccess;
+import de.nmichael.efa.data.storage.MetaData;
+import de.nmichael.efa.gui.util.TableItem;
+import de.nmichael.efa.gui.util.TableItemHeader;
+
+import java.awt.*;
+import java.util.Vector;
+
+public class EfaCloudUserRecord  extends DataRecord implements IItemListener {
+
+    // =========================================================================
+    // Field Names
+    // =========================================================================
+
+    public static final String ID               = "ID";
+    public static final String EFACLOUDUSERID   = "efaCloudUserID";
+    public static final String EMAIL            = "EMail";
+    public static final String EFAADMINNAME     = "efaAdminName";
+    public static final String VORNAME          = "Vorname";
+    public static final String NACHNAME         = "Nachname";
+    public static final String ROLLE            = "Rolle";
+    public static final String WORKFLOWS        = "Workflows";
+    public static final String CONCESSIONS      = "Concessions";
+    public static final String PASSWORT_HASH    = "Passwort_Hash";
+    public static final String USECRM           = "UseEcrm";
+
+    public static void initialize() {
+        Vector<String> f = new Vector<String>();
+        Vector<Integer> t = new Vector<Integer>();
+
+        f.add(ID);                                t.add(IDataAccess.DATA_INTEGER);
+        f.add(EFACLOUDUSERID);                    t.add(IDataAccess.DATA_INTEGER);
+        f.add(EMAIL);                             t.add(IDataAccess.DATA_STRING);
+        f.add(EFAADMINNAME);                      t.add(IDataAccess.DATA_STRING);
+        f.add(VORNAME);                           t.add(IDataAccess.DATA_STRING);
+        f.add(NACHNAME);                          t.add(IDataAccess.DATA_STRING);
+        f.add(ROLLE);                             t.add(IDataAccess.DATA_STRING);
+        f.add(WORKFLOWS);                         t.add(IDataAccess.DATA_INTEGER);
```

Comment:
As said above this is based on the PHP side of things. I understand your point, on error proneness. However, I do not expect any changes at all to this logic. It is stable since years.

Reply:
My point about changes was more related to privileges in efa being changed. When a new admin privilege CanDoThisAndThat is being introduced, it must be added to the end of the list, not the middle - otherwise the bit positions would shift and a new version would misinterpret the bitmap saved with an old version. Similar, if we were to remove a privilege, it would need to stay as a dummy field in the list since otherwise the remaining ones would shift.

So we'll have to be very careful with any privilege updates in the future.